projects/data-grid/src/lib/store/data-grid.ts

```diff
@@ -286,6 +313,26 @@ const addRow = (state: GridState, action: AddRowPayload<any>): GridState => {
     }, state);
 };
 
+const alignSelectedRowsIndexesWithRowDataIndexes = (state: GridState) => {
+  const {rowDataIndexes} = state;
+  return R.filter(index => R.includes(index, rowDataIndexes), getSelectedRowIndexes(state));
+};
+
+const calculateSelectedRowsIfCurrentPageSelected = (state: GridState) => {
+  const filteredSelectedRows = alignSelectedRowsIndexesWithRowDataIndexes(state);
+
+  return currentPageContainsSelectedRows(state)
+  ? getCurrentPageIndexes(state) : R.uniq(R.concat(getCurrentPageIndexes(state), filteredSelectedRows));
+};
+
+const recalculateSelectedRowsIndexes = (state: GridState): GridState => R.mergeRight(state,{
+    selectedRowsIndexes: R.cond([
+      [R.propEq('allPagesSelected', true), getRowDataIndexes],
+      [R.propEq('currentPageSelected', true), calculateSelectedRowsIfCurrentPageSelected],
+      [R.T, alignSelectedRowsIndexesWithRowDataIndexes]
+    ])(state)
+  });
+
 const recalculateRowIndexesAndPagination = (state: GridState): any => {
```

Comment:
we should make sure that when row indexes and pagination is recalculated, the selected rows are updated too

Reply:
I think that this will be handled with the 'updateGridRowsHandler' or i am missing something ?

Reply:
if we put all the logic there yes